.github/workflows/publish.yml

```diff
@@ -0,0 +1,28 @@
+name: CI
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+
+jobs:
+  publish:
```

Comment:
Can add a:
```yaml
tests:
    uses: ./.github/workflows/continuous-integration.yml
    secrets: inherit
```
With a `needs` on `publish` and a `on.workflow_call` on `continuous-integration.yml`

Reply:
I think it would make sense to have `publish` step executed only when the tests pass, but isn't the current version fine? I.e. the maintainer should not release a new version until all the CI jobs pass, right?

Also what is the meaning of `secrets`? I have not tried yet the trusted publishing approach, but I thought that it would not use secrets anymore?

Reply:
Yes, for the most part it is ok. Adding a dependency like that will make sure that if there were any upstream changes in the meantime, everything is still good. I usually run `ci-build-wheel` only before release because of how thorough it is and `test` just for good measure.

`secrets` is not really necessary there. It basically makes sure that your secrets (in `publish`) propagate to `continous-integration.yml`. Since these are within the same project and there are no special secret treatments, it can be dropped.